allennlp/training/trainer.py

```diff
@@ -145,7 +167,7 @@ def _save_checkpoint(self,
         if is_best:
             logger.info("Best validation performance so far. "
                         "Copying weights to %s/best.th'.", self._serialization_prefix)
-            shutil.copy(model_path, os.path.join(model_path, "best.th"))
+            shutil.copyfile(model_path, os.path.join(self._serialization_prefix, "best.th"))
```

Comment:
Should you also modify `_restore_checkpoint` to load the best model?  Or should that go somewhere else?